src/main/java/com/puppycrawl/tools/checkstyle/api/Configuration.java

```diff
@@ -62,4 +63,10 @@
      * @return unmodifiable map containing custom messages
      */
     ImmutableMap<String, String> getMessages();
+
+    /**
+     * Gets the thread mode settings.
+     * @return the thread mode settings.
+     */
+    ThreadModeSettings getThreadModeSettings();
```

Comment:
@romani This is a test only method, so I see no reason we even need it.
No production code calls it, only tests to access the private field.